src/components/ServiceFilter/ServiceFilter.tsx

```diff
@@ -0,0 +1,235 @@
+import * as React from 'react';
+import { Filter, FormControl, Toolbar } from 'patternfly-react';
+import {
+  ActiveFilter,
+  FilterType,
+  FilterValue,
+  ServiceFilterProps,
+  ServiceFilterState
+} from '../../types/ServiceFilter';
+import * as API from '../../services/Api';
+import { Namespace } from '../../types/Namespace';
+
+const serviceNameFilter: FilterType = {
+  id: 'servicename',
+  title: 'Service Name',
+  placeholder: 'Filter by Service Name',
+  filterType: 'text',
+  filterValues: []
```

Comment:
can we type filterValues or not? Array of strings?

Reply:
yes, this is part of the FilterType class, which might get two types of filtering: values entered in the input text or a select with values.
The FilterType class doesn't store the values entered by the user, but potential values in case filterType is 'select'.